### ISSUE TITLE
No softmax activation in the last layer

### ISSUE THREAD

**Issue by raphaelreinauer** [giotto-ai/giotto-deep#34]
https://github.com/giotto-ai/giotto-deep/blob/bbb382a329ea98ffa7a624c41afe1d579d108f08/gdeep/models/simple_nn.py#L43

I think we should not have any activation here since the cross-entropy loss uses the logits.

https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html
The input is expected to contain raw, unnormalized scores for each class.

It is also more efficient to compute cross-entropy directly from the logits.
